toplevel/vernac.ml

```diff
@@ -163,15 +171,16 @@ and load_vernac verbosely sid file =
      * raised, which means that we raised the end of the file being loaded *)
     while true do
       let loc, ast =
-        try Stm.parse_sentence !rsid in_pa
-        with
-        | Stm.End_of_input -> raise Stm.End_of_input
-        | any ->
-          let (e, info) = CErrors.push any in
-          let loc = Loc.get_loc info in
-          let msg = CErrors.iprint (e, info) in
-          Feedback.msg_error ?loc msg;
-          iraise (e, info)
+        (* try *)
+          Stm.parse_sentence !rsid in_pa
+        (* with *)
+        (* | any when not is_end_of_input any -> *)
+        (*   let (e, info) = CErrors.push any in *)
+        (* Re-enable when we switch back to feedback-based error printing *)
```

Comment:
This is more delicate as it is the parsing part. Different clients do different things, here the parser for load used to forward the error to the feedback handler for convenience.

But I do agree that cleaner solutions are possible, indeed I am trying not to rewrite too much but to adapt what was already there. I think at some point we should really rewrite the whole `compile` path to be cleaner, but for example I already have `sercomp` which is a new compiler.